apps/server-asset-sg/src/features/assets/files/file-processors/file-extraction.service.ts

```diff
@@ -136,4 +138,26 @@ export class FileExtractionService extends AbstractProcessingService<ExtractionR
       },
     });
   }
+
+  private async storeLanguages(fileId: AssetFileId, results: ExtractionResult[]): Promise<void> {
+    // Find all languages that have been extracted.
+    const languages = getLanguageCodesOfPages(results.map((it) => it.metadata));
+
+    // Find all assets that are mapped to the processed file.
+    const assets = await this.prisma.assetFile.findMany({
+      where: { fileId },
+      select: { assetId: true },
+    });
+
+    // For each asset, add all languages that have not yet been mapped to it.
+    await this.prisma.assetLanguage.createMany({
+      data: [...languages].flatMap((code) =>
+        assets.map(({ assetId }) => ({
+          assetId,
+          languageItemCode: code,
+        })),
```

Comment:
Wouldn't

```
assets.flatMap(({ assetId }) =>
  [...languages].map(code => ({ assetId, languageItemCode: code }))
);
```

be more readable, matching your comment? I get that we then spread the set for each asset, but we're not really talking performance here anyway.